Add database migrations documentation

djjuhasz opened the pull request.

djjuhasz committed:
Add database migrations documentation

jraddaoui commented on docs/src/dev-manual/db-migrations.md:
````diff
@@ -0,0 +1,205 @@
+```
+
+Then regenerate the `atlas.sum` hashes:
+
````
I'd like to do this in the actual migrate command appending `.up` to the passed name if it's not included already, and just add a note here for migrations created manually altogether.

djjuhasz replied:
I started trying to add the ".up" to the filename in the migrate command, but it's not trivial - you need to find the migration file name (with the prepended timestamp) then update the name of the file, update the `atlas.sum` file and regenerate the file hashes.  I decided to just document the current state as-is, with the intention of improving the migrate command later.

jraddaoui replied:
I mean adding the `.up` suffix before the migration is generated, in https://github.com/artefactual-sdps/enduro/blob/main/cmd/migrate/main.go#L108:

```go
name, _ := p.GetString("name")
if !strings.HasSuffix(name, ".up") {
	name = fmt.Sprintf("%s.up", name)
}
```

This seems to work for me and it avoids all that manual process for generated migrations.

djjuhasz replied:
Ah, that is a simple hack that does the job.  I was overthinking the problem I guess. :laughing: 

I've added this change, and updated the documentation appropriately.

jraddaoui reviewed (commented):
Nice, thanks for looking into this @djjuhasz!

djjuhasz committed:
Append ".up" to database migration name

When generating an database migration with the "migrate" command
append ".up" to the migration name. This change ensures the generated
migration file has a ".up.sql" extension, which is required when
applying the migration.

[skip-codecov]

jraddaoui reviewed (approved):
Really nice, thanks @djjuhasz!
